stl/inc/xmemory

```diff
@@ -784,6 +784,8 @@ class allocator {
 public:
     static_assert(!is_const_v<_Ty>, "The C++ Standard forbids containers of const elements "
                                     "because allocator<const T> is ill-formed.");
+    static_assert(!is_volatile_v<_Ty>, "The C++ Standard forbids containers of volatile elements "
+                                       "because allocator<volatile T> is ill-formed.");

```

Comment:
FWIW, LWG-2447 doesn't say the default allocator doesn't support cv-qualified types, it says that a type need not support cv-qualified types to be an allocator. I also believe that neither our existing "`allocator<const T>` is ill-formed" claim nor the new "`allocator<volatile T>` is ill-formed" claim is true: https://godbolt.org/z/Ed9e7KhnT.

If we think `allocator` should refuse to support cv-qualified types, I think we should at least file an LWG issue adding wording to the Standard to allow us to do so. We should probably require a non-array object type while we're at it, since the Standard currently places no requirements on `allocator`'s template type parameter.